### DISCUSSION TITLE
对service/biz/data还是有点难理解啊~

### DISCUSSION THREAD

**Discussion by xmsz** [go-kratos/kratos#1495] [General]
背景
- 之前都是MVC模型，很简单就是 controller调用和组装具体业务service，service要用到数据的时再调用model
- 转换到领域模型的时候，通过网上的教程和问答都好难理解
- 可能是因为MVC在脑海里太根深蒂固了，总是会想着去转换而不是重新认知
- 而且都说是一直概念没有具体的代码学习，就感觉更抽象了
- 还有很多词汇对我来说太过于抽象了

然后来到kratos
- 我看了go训练营关于目录的视频还有kratos博客的文章，但是因为基础不同，还是没能理解
- 我尝试看beer-shop项目，也没能理解

比如一些疑问
- biz和data好像，很多biz都是直接返回data方法，连方法名都一样
- 组装逻辑是在哪处理，我看到service也有，data也有，biz也有
- ...

大家有没有什么其他帮助理解的办法

例子的话，比如获取一个文章列表

**Comment by aliliin** [comment]
可以看看 这篇总结，[Go 工程化目录结构](https://lailin.xyz/post/go-training-week4-project-layout.html)

**Comment by Windfarer** [comment]
这个是与框架无关的非强制结构，不理解可以不用，调整回mvc模型即可

**Comment by omezro** [comment]
主要的亮点还是在wire的依赖注入上，可以先了解下wire。有时候多一个封装就是为了更灵活的调用。实在用不到的话可以自行设置回习惯的调用链。
